src/Nova-GPU-Driver.md

```diff
@@ -1,62 +1,92 @@
 # Nova GPU Driver
 
-Nova is a driver for GSP-based Nvidia GPUs that is currently under development
-and is being written in Rust.
-
-Currently, the objective is to upstream Rust abstractions for the relevant
-subsystems a prerequisite for the actual driver. Hence, the first mainline
-version of Nova will be a stub driver which helps establishing the necessary
-infrastructure in other subsystems (notably PCI and DRM).
+Nova is a driver for GSP (global system processor) based Nvidia GPUs. It is
+intended to become the successor of Nouveau as the mainline driver for Nvidia
+(GSP) GPUs on Linux.
 
 ## Contact
 
-To contact the team and / or participate in development, please use the mailing
-list: nouveau@lists.freedesktop.org
+Available communication channels are:
+
+- The mailing list: nouveau@lists.freedesktop.org
+- IRC: #nouveau on OFTC
+- [Zulip Chat](https://rust-for-linux.zulipchat.com/#narrow/channel/509436-Nova)
 
 
 ## Resources
 
-- [Official Source Tree](https://gitlab.freedesktop.org/drm/nova)
-- [Announcement E-Mail](https://lore.kernel.org/dri-devel/Zfsj0_tb-0-tNrJy@cassiopeiae/)
+The parts that are already in mainline Linux can be found in
+`drivers/gpu/nova-core/` and `drivers/gpu/drm/nova/`
+
+The out-of-tree repository where active development takes place is located on
+[Freedesktop](https://gitlab.freedesktop.org/drm/nova).
+
+
+## Background
+
+### Why a new driver?
+
+Nouveau was, for the most part, designed for pre-GSP hardware. The driver exists
+since ~2009 and its authors back in the day had to reverse engineer a lot about
+the hardware's internals, resulting in a relatively difficult to maintain
+codebase.
+
+Moreover, Nouveau's maintainers concluded that a new driver, exclusively for
+(new) GSP hardware, would allow for significantly simplifying the driver
+design.
 
-In the source tree, the driver lives in `drivers/gpu/drm/nova`.
 
+### Why write it in Rust?
 
-## Status
+Rust's most desired feature are its guarantees for memory safety, notably the
+elimination of Use-after-Free errors. Those are errors GPU drivers suffer from
```

Comment:
It's basically included in the class of problems you describe, but I would also mention that Rust's powerful type system allows us to encode a lot of logic that subsequently can be evaluated at compile time rather than runtime.

A prominent example are lifetime rules, which can be greatly evaluated at compile time, where DRM drivers in C have to enforce them by convention, which given the high complexity of DRM drivers, often leads to (memory) bugs.

Reply:
```
Besides Rust's built-in ownership and lifetime model, its powerful type system
allows us to avoid a large portion of a whole class of bugs (i.e. memory safety
bugs).

Additionally, the same features allow us to model APIs in a way that also
certain logic errors can be caught at compile time already.
```